virtwho/virt/virt.py

```diff
@@ -492,6 +497,34 @@ def _get_data(self):
             reports[source_key] = report
         return reports
 
+    def _get_data_initial(self):
+        """
+        Waits for each source in self.source_keys to have a value returned from the datastore. This
+        does not check to see if the data has been previously sent. This method will wait for a
+        maximum of the self.interval period of time, then return whatever it has gathered thus far.
+        If data for each source key is gathered before the interval has expired, this method will
+        return.
+        @return: dict
+        """
+        reports = {}
+        while not reports and not self.is_terminated():
+            source_keys_remaining = list(self.source_keys)
+            time_waited = 0
+            while len(source_keys_remaining) > 0 and time_waited < self.interval and not self.is_terminated():
+                for source_key in source_keys_remaining:
+                    try:
+                        report = self.source.get(source_key)
```

Comment:
I do not do this here because I had no interest in knowing whether the result I get back is nothing. The intention was to consider the item returned (so long as it exists) for whatever it is and consider us unblocked by that source (as we now have a value to try to send). Granted None will not result in anything being sent. Is it worth discarding None and spinning waiting for a Non-none value for all sources?